### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by quasarea** [GitTools/GitVersion#2434]
**Describe the bug**
When calling /version using azdo pipeline error gets thrown. I suspect parameters are not parsed properly and earlier parameters don't allow to run that one.

Actual command called:
```
C:\agent\_work\_tool\GitVersion.Tool\5.3.7\x64\dotnet-gitversion.exe C:/agent/_work/46/s /output json /output buildserver /config C:\agent\_work\46\s\Templates\ComponentVersioning\GitVersion.yml /version
```

## Expected Behavior
version would be displayed

## Actual Behavior
Error thrown

## Steps to Reproduce
set up a pipeline with version specific gitversion, target it to 5.3.7 and provide additional argument /version

- Version Used: 5.3.7
- Operating System and version (Windows 10, Ubuntu 18.04): Win 10

**Comment by asbjornu** [comment]
Can you try replacing `/` with `-` in the arguments?

**Comment by quasarea** [comment]
same result, replicated with 5.5.0, from tasks:
```
  - task: gitversion/setup@0
    inputs:
      versionSpec: '5.5.0'
  - task: gitversion/execute@0
    inputs:      
      additionalArguments: -version    
```
result:
```
Starting: gitversionexecute
==============================================================================
Task         : Execute GitVersion Task
Description  : Easy Semantic Versioning (http://semver.org) for projects using Git
Version      : 0.9.5
Author       : GitTools Contributors
Help         : See the [documentation](https://gitversion.net/docs/) for help
==============================================================================
Command: dotnet-gitversion C:/agent/_work/46/s /output json /output buildserver -version
C:\agent\_work\_tool\GitVersion.Tool\5.5.0\x64\dotnet-gitversion.exe C:/agent/_work/46/s /output json /output buildserver -version

Unhandled Exception: GitVersion.WarningException: Could not parse command line parameter '-version'.
##[error]SyntaxError: Unexpected end of JSON input
   at GitVersion.ArgumentParser.ParseTargetPath(Arguments arguments, String name, String[] values, String value, Boolean parseEnded) in D:\a\GitVersion\GitVersion\src\GitVersionExe\ArgumentParser.cs:line 194
   at GitVersion.ArgumentParser.ParseSwitchArguments(Arguments arguments, NameValueCollection switchesAndValues, Int32 i) in D:\a\GitVersion\GitVersion\src\GitVersionExe\ArgumentParser.cs:line 126
   at GitVersion.ArgumentParser.ParseArguments(String[] commandLineArguments) in D:\a\GitVersion\GitVersion\src\GitVersionExe\ArgumentParser.cs:line 88
   at GitVersion.Program.<>c__DisplayClass4_0.<CreateHostBuilder>b__2(IServiceProvider sp) in D:\a\GitVersion\GitVersion\src\GitVersionExe\Program.cs:line 43
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at GitVersion.GitVersionCoreModule.<>c.<RegisterTypes>b__0_0(IServiceProvider sp) in D:\a\GitVersion\GitVersion\src\GitVersionCore\GitVersionCoreModule.cs:line 55
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(IEnumerableCallSite enumerableCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at GitVersion.Program.Main(String[] args) in D:\a\GitVersion\GitVersion\src\GitVersionExe\Program.cs:line 22
   at GitVersion.Program.<Main>(String[] args)
Finishing: gitversionexecute
```

**Comment by quasarea** [comment]
It looks like recently target was updated and it broke arguments interpretation even more. Arguments that were happy running on 0.9.4 are now broken on 0.9.5.
```
Starting: GitVersion Config
==============================================================================
Task         : Execute GitVersion Task
Description  : Easy Semantic Versioning (http://semver.org) for projects using Git
Version      : 0.9.4
Author       : GitTools Contributors
Help         : See the [documentation](https://gitversion.net/docs/) for help
==============================================================================
Command: dotnet-gitversion C:/agent/_work/46/s /output json /output buildserver /config C:\agent\_work\46\s\Templates\ComponentVersioning\GitVersion.yml /targetpath C:\agent\_work\46\s\Source /showconfig
C:\agent\_work\_tool\GitVersion.Tool\5.3.7\x64\dotnet-gitversion.exe C:/agent/_work/46/s /output json /output buildserver /config C:\agent\_work\46\s\Templates\ComponentVersioning\GitVersion.yml /targetpath C:\agent\_work\46\s\Source /showconfig
INFO [10/23/20 12:11:21:12] Working directory: C:\agent\_work\46\s\Source
assembly-versioning-scheme: Major
...
```

```
Starting: GitVersion Config
==============================================================================
Task         : Execute GitVersion Task
Description  : Easy Semantic Versioning (http://semver.org) for projects using Git
Version      : 0.9.5
Author       : GitTools Contributors
Help         : See the [documentation](https://gitversion.net/docs/) for help
==============================================================================
Command: dotnet-gitversion C:/agent/_work/46/s /output json /output buildserver /config C:\agent\_work\46\s\Templates\ComponentVersioning\GitVersion.yml /targetpath C:\agent\_work\46\s\Source /showconfig
C:\agent\_work\_tool\GitVersion.Tool\5.5.0\x64\dotnet-gitversion.exe C:/agent/_work/46/s /output json /output buildserver /config C:\agent\_work\46\s\Templates\ComponentVersioning\GitVersion.yml "/targetpath C:\agent\_work\46\s\Source /showconfig"

Unhandled Exception: GitVersion.WarningException: Could not parse command line parameter '/targetpath C:\agent\_work\46\s\Source /showconfig'.
   at GitVersion.ArgumentParser.ParseTargetPath(Arguments arguments, String name, String[] values, String value, Boolean parseEnded) in D:\a\GitVersion\GitVersion\src\GitVersionExe\ArgumentParser.cs:line 194
   at GitVersion.ArgumentParser.ParseSwitchArguments(Arguments arguments, NameValueCollection switchesAndValues, Int32 i) in D:\a\GitVersion\GitVersion\src\GitVersionExe\ArgumentParser.cs:line 126
   at GitVersion.ArgumentParser.ParseArguments(String[] commandLineArguments) in D:\a\GitVersion\GitVersion\src\GitVersionExe\ArgumentParser.cs:line 88
```

**Comment by asbjornu** [comment]
This looks very wrong to me:

```
C:\agent\_work\_tool\GitVersion.Tool\5.5.0\x64\dotnet-gitversion.exe C:/agent/_work/46/s /output json /output buildserver /config C:\agent\_work\46\s\Templates\ComponentVersioning\GitVersion.yml "/targetpath C:\agent\_work\46\s\Source /showconfig"
```

Where does the quotes around `/targetpath C:\agent\_work\46\s\Source /showconfig` come from?

**Comment by quasarea** [comment]
This is what task generates from filled arguments. In that case it was:
```
  - task: gitversion/execute@0
    displayName: GitVersion Config
    inputs:
      useConfigFile: true
      configFilePath: .\Templates\ComponentVersioning\GitVersion.yml
      additionalArguments: /targetpath $(Build.SourcesDirectory)\Source /showconfig
```
if I use targetpath argument provided, so it will be:
```
  - task: gitversion/execute@0
    displayName: GitVersion Config
    inputs:
      useConfigFile: true
      configFilePath: .\Templates\ComponentVersioning\GitVersion.yml
      targetpath: $(Build.SourcesDirectory)\Source
      additionalArguments: /showconfig
```
it will still fail, but this time target path will expand to:
```
C:\agent\_work\46\s\C:\agent\_work\46\s\Source
```
I have also tried targetpath with values Source, \Source, .\Source, /Source. ./Source
None works

I think the problems with arguments interpretation comes from there, already submitted issue at https://github.com/GitTools/actions/issues/245 as today all my build are failing on things that worked three days ago 😢

**Comment by quasarea** [comment]
Fixed by  GitTools/actions#245
